node/block_exchange.go

```diff
@@ -0,0 +1,226 @@
+package node
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+
+	"github.com/celestiaorg/go-header"
+	goheaderp2p "github.com/celestiaorg/go-header/p2p"
+	goheaderstore "github.com/celestiaorg/go-header/store"
+	goheadersync "github.com/celestiaorg/go-header/sync"
+	ds "github.com/ipfs/go-datastore"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/p2p/net/conngater"
+	"github.com/tendermint/tendermint/libs/log"
+	tmtypes "github.com/tendermint/tendermint/types"
+	"go.uber.org/multierr"
+
+	"github.com/rollkit/rollkit/config"
+	"github.com/rollkit/rollkit/p2p"
+	"github.com/rollkit/rollkit/types"
+)
+
+type BlockExchangeService struct {
```

Comment:
We should get in the habit of better documentation of our structs. We should explain what the struct is responsible for so that developer have the right context to be able to understand why it has the fields it does. 

This then allows us to add more helpful comments on the fields to explain what they are responsible for. 

Especially with a project like rollkit that is meant to support so many different DAs, EEs, rollup types, etc, I think over documenting our code is going to help us in the long run in terms of catching edge cases and understanding the impact of design changes.